lib/scholar/naive_bayes/bernoulli.ex

```diff
@@ -0,0 +1,407 @@
+defmodule Scholar.NaiveBayes.Bernoulli do
+  @moduledoc """
+  Naive Bayes classifier for multivariate Bernoulli models.
+  Like MultinomialNB, this classifier is suitable for discrete data. The
+  difference is that while MultinomialNB works with occurrence counts,
+  BernoulliNB is designed for binary/boolean features.
+  """
+  import Nx.Defn
+  import Scholar.Shared
+
+  @derive {Nx.Container,
+           containers: [
+             :feature_count,
+             :class_count,
+             :class_log_priors,
+             :feature_log_probability
+           ]}
+  defstruct [:feature_count, :class_count, :class_log_priors, :feature_log_probability]
+
+  opts_schema = [
+    num_classes: [
+      type: :pos_integer,
+      required: true,
+      doc: ~S"""
+      Number of different classes used in training.
+      """
+    ],
+    alpha: [
+      type: {:or, [:float, {:list, :float}]},
+      default: 1.0,
+      doc: ~S"""
+      Additive (Laplace/Lidstone) smoothing parameter
+      (set alpha to 0.0 and force_alpha to true, for no smoothing).
+      """
+    ],
+    force_alpha: [
+      type: :boolean,
+      default: true,
+      doc: ~S"""
+      If `false` and alpha is less than 1e-10, it will set alpha to
+      1e-10. If `true`, alpha will remain unchanged. This may cause
+      numerical errors if alpha is too close to 0.
+      """
+    ],
+    binarize: [
+      type: {:or, [:float, {:in, [nil]}]},
+      default: 0.0,
+      doc: ~S"""
+      Threshold for binarizing (mapping to booleans) of sample features.
+      If nil, input is presumed to already consist of binary vectors.
+      """
+    ],
+    fit_priors: [
+      type: :boolean,
+      default: true,
+      doc: ~S"""
+      Whether to learn class prior probabilities or not.
+      If `false`, a uniform prior will be used.
+      """
+    ],
+    class_priors: [
+      type: {:custom, Scholar.Options, :weights, []},
+      doc: ~S"""
+      Prior probabilities of the classes. If specified, the priors are not
+      adjusted according to the data.
+      """
+    ],
+    sample_weights: [
+      type: {:custom, Scholar.Options, :weights, []},
+      doc: ~S"""
+      List of `num_samples` elements.
+      A list of 1.0 values is used if none is given.
+      """
+    ]
+  ]
+
+  @opts_schema NimbleOptions.new!(opts_schema)
+
+  @doc """
+  Fits a naive Bayes model. The function assumes that the targets `y` are integers
+  between 0 and `num_classes` - 1 (inclusive). Otherwise, those samples will not
+  contribute to `class_count`.
+  ## Options
```

Comment:
Please remember to add new lines before and after each section header. Before lists and so on. :)